src/Rules/Program.hs

```diff
@@ -19,15 +20,16 @@ import Target
 import UserSettings
 import Util
 
--- | Wrapper is an expression depending on the 'FilePath' to the wrapped binary.
-type Wrapper = FilePath -> Expr String
+-- | Wrapper is an expression depending on the 'FilePath' to the
+-- | library path and name of the wrapped binary.
+type Wrapper = FilePath -> FilePath -> Expr String
```

Comment:
Right, I see. So, we could use `WrappedBinary { binaryLibPath :: FilePath, binaryName :: FilePath }` to make it easier to get the order of arguments right.